### ISSUE TITLE
History sync_on_enter=false does not work with sqlite

### ISSUE THREAD

**Issue by sam701** [nushell/nushell#7915]
### Describe the bug

When`config.history.file_format=sqlite`, `sync_on_enter=false` does not work. The history is always synced on enter.

### How to reproduce

1. Configure shell
```
  history: {
    sync_on_enter: false
    file_format: "sqlite"
  }
```
2. Open a new shell and type `ls aaa`
3. Open another shell and start typing `ls`
4. The auto-completion is shown: `ls aaa`

### Expected behavior

`ls aaa` is not visible in the newly open shell because the previous shell should not have synchronized its history.

### Configuration

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.74.0                                                            |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | macos-x86_64                                                      |
| build_target       | x86_64-apple-darwin                                               |
| rust_version       | rustc 1.66.0 (69f9c33d7 2022-12-12) (built from a source tarball) |
| cargo_version      | cargo 1.66.0 (d65d197ad 2022-11-15)                               |
| pkg_version        | 0.74.0                                                            |
| build_time         | 2023-01-10 20:12:41 +00:00                                        |
| build_rust_channel | release                                                           |
| features           | database, default, trash, which, zip                              |
| installed_plugins  |                                                                   |

**Comment by fdncred** [comment]
Please help me understand why you'd want this behavior. I don't think `sync_on_enter` was ever supposed to be used with sqlite history.

I assume you want to have unique histories per terminal window. If that's the case, then `history session` needs to be used to only show the history items unique to that session. Unfortunately, we don't have an out-of-the-box example but one could be constructed.

**Comment by sholderbach** [comment]
> I assume you want to have unique histories per terminal window. If that's the case, then `history session` needs to be used to only show the history items unique to that session. Unfortunately, we don't have an out-of-the-box example but one could be constructed.

I don't think we should expect the users to do some manual filtering crimes in their history menu etc. just to achieve this. 

> Please help me understand why you'd want this behavior. I don't think `sync_on_enter` was ever supposed to be used with sqlite history.


I think at the moment it is safe to say that `sync_on_enter: false` is generallly broken/useless (see the issue with exiting behavior for the file)

**Comment by fdncred** [comment]
> I don't think we should expect the users to do some manual filtering crimes in their history menu etc. just to achieve this.

I tend to agree, but for now, this is the only alternative we have until someone contributes a way to do this. I already asked phiresky to help with this but he wasn't able to respond. To be clear, I don't think there's any way to use `history session` to show per session hints, but I think it's possible to be used to show per-history items with some keybinding like ctrl+r.

> I think at the moment it is safe to say that sync_on_enter: false is generally broken

agreed.
